### ISSUE TITLE
🐛 [cloud_functions] build error - Requested but did not find extension point with identifier Xcode.IDEKit.ExtensionSentinelHostApplications for extension Xcode.DebuggerFoundation.AppExtensionHosts.watchOS of plug-in com.apple.dt.IDEWatchSupportCore

### ISSUE THREAD

**Issue by StevenJPx2** [firebase/flutterfire#9003]
## Bug report

**Describe the bug**
Build fails when cloud_functions is installed, but runs fine in dev.

### Steps to reproduce

1. `flutter pub add cloud_functions`
2. Change minimum platform version in `Podfile`
3. `flutter build ios --debug --no-codesign`
4. See error or incorrect behavior

### Expected behavior

<details><summary>Error</summary>

```
Warning: Building for device with codesigning disabled. You will have to manually codesign before deploying to device.
Building com.example.appFlutter for device (ios)...
Running Xcode build...                                          
 └─Compiling, linking and signing...                        498ms
Xcode build done.                                            4.6s
Failed to build iOS app
Error output from Xcode build:
↳
    2022-06-30 14:16:18.023 xcodebuild[75784:4155007] Requested but did not find extension point with identifier Xcode.IDEKit.ExtensionSentinelHostApplications for extension Xcode.DebuggerFoundation.AppExtensionHosts.watchOS of plug-in com.apple.dt.IDEWatchSupportCore
    2022-06-30 14:16:18.023 xcodebuild[75784:4155007] Requested but did not find extension point with identifier Xcode.IDEKit.ExtensionPointIdentifierToBundleIdentifier for extension Xcode.DebuggerFoundation.AppExtensionToBundleIdentifierMap.watchOS of plug-in com.apple.dt.IDEWatchSupportCore
    ** BUILD FAILED **


Xcode's output:
↳
    Writing result bundle at path:
    	/var/folders/tg/n2ghb0wx6x3flzwbvg_t32rw0000gn/T/flutter_tools.gdqquU/flutter_ios_build_temp_dirI55p0Z/temporary_xcresult_bundle

    ld: framework not found Pods_Runner
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    ld: framework not found Pods_Runner
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    error: the following command failed with exit code 1 but produced no further output
    Ld /Users/***/Library/Developer/Xcode/DerivedData/Runner-awsimrqjwnrsiodqoxooalbhinzm/Build/Intermediates.noindex/Runner.build/Debug-iphoneos/Runner.build/Objects-normal/arm64/Binary/Runner normal arm64 (in target 'Runner' from project 'Runner')
    note: Using new build system
    note: Planning
    note: Build preparation complete
    note: Building targets in dependency order

    Result bundle written to path:
    	/var/folders/tg/n2ghb0wx6x3flzwbvg_t32rw0000gn/T/flutter_tools.gdqquU/flutter_ios_build_temp_dirI55p0Z/temporary_xcresult_bundle



```

</details>
---

## Additional context

<details><summary><code>ios/Podfile</code></summary>

```Podfile
# Uncomment this line to define a global platform for your project
platform :ios, '12.0' # 13 gives the same error

# CocoaPods analytics sends network stats synchronously affecting flutter build latency.
ENV['COCOAPODS_DISABLE_STATS'] = 'true'

project 'Runner', {
  'Debug' => :debug,
  'Profile' => :release,
  'Release' => :release,
}

def flutter_root
  generated_xcode_build_settings_path = File.expand_path(File.join('..', 'Flutter', 'Generated.xcconfig'), __FILE__)
  unless File.exist?(generated_xcode_build_settings_path)
    raise "#{generated_xcode_build_settings_path} must exist. If you're running pod install manually, make sure flutter pub get is executed first"
  end

  File.foreach(generated_xcode_build_settings_path) do |line|
    matches = line.match(/FLUTTER_ROOT\=(.*)/)
    return matches[1].strip if matches
  end
  raise "FLUTTER_ROOT not found in #{generated_xcode_build_settings_path}. Try deleting Generated.xcconfig, then run flutter pub get"
end

require File.expand_path(File.join('packages', 'flutter_tools', 'bin', 'podhelper'), flutter_root)

flutter_ios_podfile_setup

target 'Runner' do
  use_frameworks!
  use_modular_headers!

  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
  end
end

```

</details>

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
[✓] Flutter (Channel stable, 3.0.3, on macOS 12.4 21F79 darwin-arm, locale en-IN)
    • Flutter version 3.0.3 at /Users/***/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 676cefaaff (8 days ago), 2022-06-22 11:34:49 -0700
    • Engine revision ffe7b86a1e
    • Dart version 2.17.5
    • DevTools version 2.12.2

[!] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/***/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/***/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses

[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.58.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.25.0

[✓] Connected device (3 available)
    • iPhone 13 Pro (mobile) • 34CAD4B2-0294-434C-87C9-82AA52689DA6 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-5 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64   • macOS 12.4 21F79 darwin-arm
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 103.0.5060.53

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
Dart SDK 2.17.5
Flutter SDK 3.0.3
app_flutter 1.0.0+1

dependencies:
- cloud_functions 3.2.17 [cloud_functions_platform_interface cloud_functions_web firebase_core firebase_core_platform_interface flutter]
- cupertino_icons 1.0.4
- firebase_core 1.18.0 [firebase_core_platform_interface firebase_core_web flutter meta]
- flutter 0.0.0 [characters collection material_color_utilities meta vector_math sky_engine]
- flutter_svg 1.0.3 [flutter meta path_drawing vector_math xml]
- pin_code_fields 7.4.0 [flutter]

dev dependencies:
- flutter_lints 1.0.4 [lints]
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher material_color_utilities meta source_span stream_channel string_scanner term_glyph]

transitive dependencies:
- async 2.8.2 [collection meta]
- boolean_selector 2.1.0 [source_span string_scanner]
- characters 1.2.0
- charcode 1.3.1
- clock 1.1.0
- cloud_functions_platform_interface 5.1.8 [firebase_core flutter meta plugin_platform_interface]
- cloud_functions_web 4.2.16 [cloud_functions_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins js]
- collection 1.16.0
- fake_async 1.3.0 [clock collection]
- firebase_core_platform_interface 4.4.1 [collection flutter meta plugin_platform_interface]
- firebase_core_web 1.6.5 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- flutter_web_plugins 0.0.0 [flutter js characters collection material_color_utilities meta vector_math]
- js 0.6.4
- lints 1.0.1
- matcher 0.12.11 [stack_trace]
- material_color_utilities 0.1.4
- meta 1.7.0
- path 1.8.1
- path_drawing 1.0.0 [vector_math meta path_parsing flutter]
- path_parsing 1.0.0 [vector_math meta]
- petitparser 4.4.0 [meta]
- plugin_platform_interface 2.1.2 [meta]
- sky_engine 0.0.99
- source_span 1.8.2 [collection path term_glyph]
- stack_trace 1.10.0 [path]
- stream_channel 2.1.0 [async]
- string_scanner 1.1.0 [charcode source_span]
- term_glyph 1.2.0
- test_api 0.4.9 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph matcher]
- vector_math 2.1.2
- xml 5.3.1 [collection meta petitparser]
```

</details>

---

**Comment by StevenJPx2** [comment]
Additionally, I have tried the proposed solution in [#8984](https://github.com/firebase/flutterfire/issues/8984#issuecomment-1168481569) and that doesn't work either.

Error:
```
Warning: Building for device with codesigning disabled. You will have to manually codesign before deploying to device.
Building com.example.appFlutter for device (ios)...
Running pod install...                                             331ms
CocoaPods' output:
↳

    [!] Invalid `Podfile` file: wrong number of arguments (given 0, expected 1..2).

     #  from ***/ios/Podfile:37
     #  -------------------------------------------
     #  
     >  target.build_configurations.each do |config|
     #    config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '12.0'
     #  -------------------------------------------

    /Library/Ruby/Gems/2.6.0/gems/cocoapods-core-1.11.3/lib/cocoapods-core/podfile.rb:335:in `rescue in block in from_ruby'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-core-1.11.3/lib/cocoapods-core/podfile.rb:329:in `block in from_ruby'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-core-1.11.3/lib/cocoapods-core/podfile.rb:50:in `instance_eval'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-core-1.11.3/lib/cocoapods-core/podfile.rb:50:in `initialize'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-core-1.11.3/lib/cocoapods-core/podfile.rb:327:in `new'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-core-1.11.3/lib/cocoapods-core/podfile.rb:327:in `from_ruby'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-core-1.11.3/lib/cocoapods-core/podfile.rb:293:in `from_file'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.11.3/lib/cocoapods/config.rb:205:in `podfile'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.11.3/lib/cocoapods/command.rb:160:in `verify_podfile_exists!'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.11.3/lib/cocoapods/command/install.rb:46:in `run'
    /Library/Ruby/Gems/2.6.0/gems/claide-1.1.0/lib/claide/command.rb:334:in `run'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.11.3/lib/cocoapods/command.rb:52:in `run'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.11.3/bin/pod:55:in `<top (required)>'
    /usr/local/bin/pod:23:in `load'
    /usr/local/bin/pod:23:in `<main>'
```

**Comment by StevenJPx2** [comment]
I'll try with 11, have you tried it with 12 or 13?

**Comment by darshankawar** [comment]
Yes, same with either of those versions. Maybe check if any of the dependencies in your `pubspec.yaml` could be causing the conflict or not. Check also by `flutter clean` and then running the build command.
You can also try by creating new flutter project and follow steps to replicate mentioned above and see if it still gives you same error or not.

**Comment by StevenJPx2** [comment]
I don’t know if this brings more clarity, but before this, I ran `flutterfire configure` and installed firebase_core.

**Comment by darshankawar** [comment]
Looking at the error log you shared earlier, it indicates a linker error. So since you changed platform in your podfile file, also make sure to update iOS Deployment Target in Pods target.

Also the platform ios version you set in podfile, make sure it matches with the pods target in xcode, If there's a mismatch or higher or lower version, then the said error could arise.

**Comment by berbsd** [comment]
I got myself out of this situation. Hopefully this will help others. 

My project started to fail to run on device when running from VSCode or Android Studio, but I noticed it was working fine when started from Xcode. It may have to do with Xcode knowing how to prepare the watch and phone.  so

1 - build your project in Xcode, keep Xcode open
2 - switch to VSCode or Android Studio and proceed as usual

These steps above fixed it for me. Good luck everyone and maybe someone will get to the bottom of the problem. 

TL:DR - I started to run into the same issue on my project last night - I thought it was linked to the upgrade to firebase 9.2.0 but I started to notice that even the basic "flutter create" template failed to build. Reinstalling everything down to flutter did nothing to help.

**Comment by StevenJPx2** [comment]
@darshankawar It failed with completely clean project with this error:

<details>
<summary>Error:</summary>

```
arning: Building for device with codesigning disabled. You will have to manually codesign before deploying to device.
Building com.example.testFirebase for device (ios)...
Running pod install...                                             51.4s
Running Xcode build...
 └─Compiling, linking and signing...                         3.3s
Xcode build done.                                           27.6s
Failed to build iOS app
Error output from Xcode build:
↳
    2022-07-05 15:01:42.740 xcodebuild[18218:1893777] Requested but did not find extension point with identifier Xcode.IDEKit.ExtensionSentinelHostApplications for extension Xcode.DebuggerFoundation.AppExtensionHosts.watchOS of plug-in com.apple.dt.IDEWatchSupportCore
    2022-07-05 15:01:42.741 xcodebuild[18218:1893777] Requested but did not find extension point with identifier Xcode.IDEKit.ExtensionPointIdentifierToBundleIdentifier for extension Xcode.DebuggerFoundation.AppExtensionToBundleIdentifierMap.watchOS of plug-in
    com.apple.dt.IDEWatchSupportCore
    ** BUILD FAILED **


Xcode's output:
↳
    Writing result bundle at path:
    	/var/folders/tg/n2ghb0wx6x3flzwbvg_t32rw0000gn/T/flutter_tools.SKQ1Xe/flutter_ios_build_temp_dirBseA8c/temporary_xcresult_bundle

    /***/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_core-1.19.1/ios/Classes/FLTFirebaseCorePlugin.m:229:50: warning: incompatible pointer to integer conversion sending 'NSNumber * _Nonnull __strong' to parameter of type 'BOOL' (aka 'signed char')
    [-Wint-conversion]
        [firebaseApp setDataCollectionDefaultEnabled:enabled];
                                                     ^~~~~~~
    In module 'FirebaseCore' imported from /***/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_core-1.19.1/ios/Classes/FLTFirebasePlugin.h:10:
    /***/test_firebase/ios/Pods/FirebaseCore/FirebaseCore/Sources/Public/FirebaseCore/FIRApp.h:125:10: note: passing argument to parameter 'dataCollectionDefaultEnabled' here
        BOOL dataCollectionDefaultEnabled;
             ^
    1 warning generated.
    /***/flutter/.pub-cache/hosted/pub.dartlang.org/cloud_functions-3.3.1/ios/Classes/FLTFirebaseFunctionsPlugin.m:51:71: warning: values of type 'NSInteger' should not be used as format arguments; add an explicit cast to 'long' instead [-Wformat]
            NSString *httpsErrorCode = [NSString stringWithFormat:@"%ld", error.code];
                                                                    ~~~   ^~~~~~~~~~
                                                                    %ld   (long)
    1 warning generated.
    ld: framework not found Pods_Runner
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    note: Using new build system
    note: Planning
    note: Build preparation complete
    note: Building targets in dependency order

    Result bundle written to path:
    	/var/folders/tg/n2ghb0wx6x3flzwbvg_t32rw0000gn/T/flutter_tools.SKQ1Xe/flutter_ios_build_temp_dirBseA8c/temporary_xcresult_bundle


════════════════════════════════════════════════════════════════════════════════
Building a deployable iOS app requires a selected Development Team with a
Provisioning Profile. Please ensure that a Development Team is selected by:
  1- Open the Flutter project's Xcode target with
       open ios/Runner.xcworkspace
  2- Select the 'Runner' project in the navigator then the 'Runner' target
     in the project settings
  3- Make sure a 'Development Team' is selected under Signing & Capabilities > Team.
     You may need to:
         - Log in with your Apple ID in Xcode first
         - Ensure you have a valid unique Bundle ID
         - Register your device with your Apple Developer Account
         - Let Xcode automatically provision a profile for your app
  4- Build or run your project again

For more information, please visit:
  https://flutter.dev/docs/get-started/install/macos#deploy-to-ios-devices

Or run on an iOS simulator without code signing
════════════════════════════════════════════════════════════════════════════════
Encountered error while building for device.
```
</details>

### Steps to reproduce
1. `flutter create <project-name>`
2. `cd <project-name>`
3. `flutter pub get`
4. `flutterfire configure`
5. `flutter pub add firebase_core cloud_functions`
6. `flutter clean`
7. Change version number on the Podfile
8. `flutter build ios --debug --no-codesign`

**Comment by darshankawar** [comment]
From your latest error log, it seems to be related to `firebase_core` plugin changes.
Can you try the latest version of `firebase_core_platform_interface` https://pub.dev/packages/firebase_core_platform_interface/changelog and see if it helps ?

Also, try [this](https://github.com/firebase/flutterfire/issues/9015#issuecomment-1173604569) solution and see if it works for you.

**Comment by StevenJPx2** [comment]
Neither worked in my case and give the exact same error. I imagine that it is because the pod that @russellwheatley suggested in there was for `firestore` and not for `cloud_functions`.

**Comment by darshankawar** [comment]
> ════════════════════════════════════════════════════════════════════════════════
> Building a deployable iOS app requires a selected Development Team with a
> Provisioning Profile. Please ensure that a Development Team is selected by:
>   1- Open the Flutter project's Xcode target with
>        open ios/Runner.xcworkspace
>   2- Select the 'Runner' project in the navigator then the 'Runner' target
>      in the project settings
>   3- Make sure a 'Development Team' is selected under Signing & Capabilities > Team.
>      You may need to:
>          - Log in with your Apple ID in Xcode first
>          - Ensure you have a valid unique Bundle ID
>          - Register your device with your Apple Developer Account
>          - Let Xcode automatically provision a profile for your app
>   4- Build or run your project again
> 
> For more information, please visit:
>   https://flutter.dev/docs/get-started/install/macos#deploy-to-ios-devices
> 
> Or run on an iOS simulator without code signing
> ════════════════════════════════════════════════════════════════════════════════
> Encountered error while building for device.

Have you confirmed this is setup properly as indicated already for your project ?

**Comment by StevenJPx2** [comment]
Fixed the issue myself. 

## Steps I used to fix the issue
1. Create duplicate of `ios/`
2. `flutter create .` in the root directory.
3. Copied Info.plist
